data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java

```diff
@@ -72,6 +77,20 @@ public class StreamWorker {
     private final JsonWriterSettings writerSettings = JsonWriterSettings.builder()
             .outputMode(JsonMode.RELAXED)
             .objectIdConverter((value, writer) -> writer.writeString(value.toHexString()))
+            .binaryConverter((value, writer) ->  writer.writeString(Base64.getEncoder().encodeToString(value.getData())))
+            .dateTimeConverter((value, writer) -> writer.writeNumber(String.valueOf(value.longValue())))
+            .decimal128Converter((value, writer) -> writer.writeString(value.bigDecimalValue().toPlainString()))
+            .doubleConverter((value, writer) -> writer.writeNumber(String.valueOf(value.doubleValue())))
+            .maxKeyConverter((value, writer) -> writer.writeNull())
+            .minKeyConverter((value, writer) -> writer.writeNull())
+            .regularExpressionConverter((value, writer) -> {
+                writer.writeStartObject();
+                writer.writeString(REGEX_PATTERN, value.getPattern());
+                writer.writeString(REGEX_OPTIONS, value.getOptions());
+                writer.writeEndObject();
+            })
+            .timestampConverter((value, writer) -> writer.writeNumber(String.valueOf(value.getValue())))
```

Comment:
Again, can we avoid `String.valueOf`?

```suggestion
            .timestampConverter((value, writer) -> writer.writeNumber(value.getValue()))
```

Reply:
`writer.writeNumber` only accepts String.